sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/fileSourceInterfaces.scala

```diff
@@ -298,6 +309,28 @@ trait FileFormat {
 }
 
 /**
+ * The base class file format that is based on text file.
+ */
+abstract class TextBasedFileFormat extends FileFormat {
+  private var codecFactory: CompressionCodecFactory = null
+  override def isSplitable(
+      sparkSession: SparkSession,
+      options: Map[String, String],
+      path: Path): Boolean = {
+    if (codecFactory == null) {
+      synchronized {
+        if (codecFactory == null) {
+          codecFactory = new CompressionCodecFactory(
```

Comment:
sorry. It seems we can use `sparkSession.sessionState.newHadoopConf()` instread of `sparkSession.sessionState.newHadoopConfWithOptions(options)` (I checked  the `FileSourceStrategySuite` test passed without passing `options` in `CompressionCodecFactory`).
So, we need `options` in the arguments of `isSplitable`?

```
        if (codecFactory == null) {
          codecFactory = new CompressionCodecFactory(sparkSession.sessionState.newHadoopConf())
        }
```